### PR TITLE
New Tutorial

### PR DESCRIPTION
**Pull request by qbzzt**
Replace the existing tutorial with one that is more interactive and up to date.

### CODE REVIEW

File: hardhat/README.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+[![Discord](https://img.shields.io/discord/667044843901681675.svg?color=768AD4&label=discord&logo=https%3A%2F%2Fdiscordapp.com%2Fassets%2F8c9701b98ad4372b58f13fd9f65f966e.svg)](https://discord.com/channels/667044843901681675)
+[![Twitter Follow](https://img.shields.io/twitter/follow/optimismPBC.svg?label=optimismPBC&style=social)](https://twitter.com/optimismPBC)
+
+This tutorial aims to help you get started with developing decentralized applications on [Optimism](https://optimism.io/) using HardHat. Applications 
+running on top of Optimism are about as secure as those running on the underlying Ethereum mainnet itself, but are
+[significantly cheaper](https://optimism.io/gas-comparison).
+
+
+## Build an Optimism Server
```
Comment (smartcontracts) [inline hardhat/README.md:11]: Maybe:

```suggestion
## Build an Optimistic Ethereum Node
```

File: hardhat/README.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+  out of date, so we'll get the package from a different source.
+
+   ```sh
+   curl -sL https://deb.nodesource.com/setup_12.x -o nodesource_setup.sh
+   sudo bash nodesource_setup.sh
+   sudo apt install -y nodejs
+   sudo npm install -g yarn hardhat
+   ```
+   
+3. Start a new terminal window to ensure the OS knows you are now a member of the `docker`
```
Comment (smartcontracts) [inline hardhat/README.md:40]: Can probably leave out the stuff about being a member of the `docker` group. Maybe simpler to say "log out and log back in again" or similar.

File: hardhat/README.md [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+This process downloads, compiles, and builds an Optimism network. Note that it takes a long time.
+
+```sh
+git clone https://github.com/ethereum-optimism/optimism.git
+cd optimism
+yarn install
+yarn build
+cd ops
+docker-compose build
```
Comment (smartcontracts) [inline hardhat/README.md:54]: You can speed this process up quite a bit by adding:

```
export COMPOSE_DOCKER_CLI_BUILD=1
export DOCKER_BUILDKIT=1
```

before `docker-compose build`.

Alternatively, `docker-compose build --parallel` is also quite a bit faster.

File: truffle/README.md [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+[click here](https://github.com/ethereum-optimism/optimism-tutorial/). The 
+`truffle/dapp` directory
+is just an `npm install` away from being a working example.
+
+### Get a Sample Application
+
+The easiest way is to start with a sample application. 
+
+1. Open a second command line terminal
+2. Run `hardhat`, the development environment we use in this tutorial
```
Comment (smartcontracts) [inline truffle/README.md:82]: Seems like this is still using the content for `hardhat` instead of `truffle`

**Review by smartcontracts — COMMENTED** [review]
Leaving some initial review. It seems the truffle stuff is using content from hardhat. If this PR is still a work in progress I would recommend turning this into a draft PR (search for the "Still in progress? Convert to draft" button under the reviewers section).

File: truffle/README.md (hunk 4 above)
Comment (qbzzt) [reply truffle/README.md:87]: Yes, I forgot I already submitted the PR. I'll move it to a different branch.

File: hardhat/README.md (hunk 3 above)
Comment (qbzzt) [reply hardhat/README.md:54]: Is there a reason not to do both?

Comment (smartcontracts) [reply hardhat/README.md:54]: --parallel will not work if the first is enabled (docker won't listen to it I think). TBH I am not sure which is faster. I can test it if you'd like.

**Pull request closed without merging**